# Configuration for Controller (29DOF version)
control_dt: 0.02

msg_type: "hg"     # "hg" or "go"
imu_type: "pelvis"    # "torso" or "pelvis"

lowcmd_topic: "/lowcmd"
lowstate_topic: "/lowstate"

# Model paths
policy_path: "/path/to/policy.pt"  # Not used in ONNX version
onnx_path: "policy.onnx"
motion_path: "motion.npz"


# Control gains for 29DOF (stiffness)
kps: [40.17923847137318, 99.09842777666113, 40.17923847137318, 99.09842777666113, 28.50124619574858, 28.50124619574858,
      40.17923847137318, 99.09842777666113, 40.17923847137318, 99.09842777666113, 28.50124619574858, 28.50124619574858,
      40.17923847137318, 28.50124619574858, 28.50124619574858, 14.25062309787429, 14.25062309787429, 14.25062309787429,
      14.25062309787429, 14.25062309787429, 16.77832748089279, 16.77832748089279, 14.25062309787429, 14.25062309787429,
      14.25062309787429, 14.25062309787429, 14.25062309787429, 16.77832748089279, 16.77832748089279]

# Control gains for 29DOF (damping)
kds: [2.5578897650279457, 6.3088018534966395, 2.5578897650279457, 6.3088018534966395, 1.814445686584846, 1.814445686584846,
      2.5578897650279457, 6.3088018534966395, 2.5578897650279457, 6.3088018534966395, 1.814445686584846, 1.814445686584846,
      2.5578897650279457, 1.814445686584846, 1.814445686584846, 0.907222843292423, 0.907222843292423, 0.907222843292423,
      0.907222843292423, 0.907222843292423, 1.06814150219, 1.06814150219, 0.907222843292423, 0.907222843292423,
      0.907222843292423, 0.907222843292423, 0.907222843292423, 1.06814150219, 1.06814150219]

# Default positions for 29DOF
default_angles: [-0.312, 0.0, 0.0, 0.669, -0.363, 0.0,
                 -0.312, 0.0, 0.0, 0.669, -0.363, 0.0,
                 0.0, 0.0, 0.0, 
                 0.2, 0.2, 0.0, 0.6, 0.0, 0.0, 0.0, 
                 0.2, -0.2,0.0, 0.6, 0.0, 0.0, 0.0]

# Action scaling for 29DOF
action_scale: [0.5475464652142303, 0.3506614663788243, 0.5475464652142303, 0.3506614663788243, 0.43857731392336724, 0.43857731392336724,
               0.5475464652142303, 0.3506614663788243, 0.5475464652142303, 0.3506614663788243, 0.43857731392336724, 0.43857731392336724,
               0.5475464652142303, 0.43857731392336724, 0.43857731392336724, 0.43857731392336724, 0.43857731392336724, 0.43857731392336724,
               0.43857731392336724, 0.43857731392336724, 0.07450087032950714, 0.07450087032950714, 0.43857731392336724, 0.43857731392336724,
               0.43857731392336724, 0.43857731392336724, 0.43857731392336724, 0.07450087032950714, 0.07450087032950714]

# Model dimensions
num_actions: 29
num_obs: 428  # 290 + 15 + 30 + 3 + 3 + 29 + 29 + 29 = 428

# Joint names for 29DOF (mujoco order)
joint_names: ["left_hip_pitch_joint", "left_hip_roll_joint", "left_hip_yaw_joint", "left_knee_joint", "left_ankle_pitch_joint", "left_ankle_roll_joint",
              "right_hip_pitch_joint", "right_hip_roll_joint", "right_hip_yaw_joint", "right_knee_joint", "right_ankle_pitch_joint", "right_ankle_roll_joint",
              "waist_yaw_joint", "waist_roll_joint", "waist_pitch_joint", "left_shoulder_pitch_joint", "left_shoulder_roll_joint", "left_shoulder_yaw_joint",
              "left_elbow_joint", "left_wrist_roll_joint", "left_wrist_pitch_joint", "left_wrist_yaw_joint", "right_shoulder_pitch_joint", "right_shoulder_roll_joint",
              "right_shoulder_yaw_joint", "right_elbow_joint", "right_wrist_roll_joint", "right_wrist_pitch_joint", "right_wrist_yaw_joint"]
